### DISCUSSION TITLE
I wanted to observe how vibrant is the LXQt forum before I download and install the Debian LXQt Live ISO

### DISCUSSION THREAD

**Discussion by rajibando** [lxqt/lxqt#2434] [Q&A]
Hello, my dear illustrious Leaders and LXQt Discussion Forum maintainers,
If this is not the one and only and the original discussion forum for LXQt users, then may I please be gently led to the link that is the Discussion forum for LXQt.
To me it appeared that LXDE and OpenBox is as good as dead, going by their respective Forum and Mailing List respectively. Is there a fork of OpenBox that is available to be used as a WM for LXQt users, when the DE is not very much required?
Could I please be helped in this regard?
Best wishes,
Rajib

**Comment by tsujan** [comment]
> Is there a fork of OpenBox that is available to be used as a WM for LXQt users

Any DE-agnostic X11 WM can be used with LXQt, regardless of which WM your distro may provide "by default". Among stacking WMs, I guess Openbox, xfwm4 and KWin are mostly used by LXQt users. I've also seen many LXQt users that have tiling WMs.

Openbox's development stopped years ago, but it's still usable. Its bugs are mostly related to multi-screen setups. So, if you don't have multiple monitors, you might not encounter a serious problem with it. It doesn't have a compositor.

xfwm4 is light and quite advanced. It has its own compositor (other compositors should be disabled if you use a WM that comes with a compositor). It's being actively developed by XFCE team.

KWin (my favorite) is very advanced and Qt-friendly. However, it may have too many KDE dependencies for some users. Also, it has been said that KWin is "heavy"; I can't confirm that, but it may depend on factors like graphics card or wrong settings.

**Reply by rajibando** [reply]
Thank you for your reply to my post.

Do you mean the following scenario by "stacking WMs": Start Menu ⟶ Sub-menus ⟶ possibly further sub-sub-menus) ⟶ A particular Program Link with every link on the list? That is, a Cascading Menu System beginning with Start Menu ⟶ Sub-menus ⟶ possibly further Sub-sub-menus ⟶ A particular Program Link with every link on the list?

This makes sense when you talk about Tiling WMs.

Does xfwm accompany xfce or just related to xfce? Does KWin accompany KDE or just related to KDE?

What is a compositor that you have mentioned in respect of OpenBox?

I hope that I could carry this conversation forward with you without interference until I have learnt substantially for the objective I wish to fulfil.

On [my other post](https://github.com/lxqt/lxqt/discussions/2432#discussioncomment-6056143) in the thread, [Modify right-click desktop menu?](https://github.com/lxqt/lxqt/discussions/2432) I have posted a link to a post in the Debian Forum on that issue, and I want to actually learn how to add a custom menu-entry in _pcmanfm_ linking to a program, script or a command, by right-clicking on the clear portion or an icon within pcmanfm window, for which I have received no solution.

After visiting your post I have added a post to the said thread on the [Debian Mailing List with a confidence that I would be able to post a solution for the said problem shortly](https://lists.debian.org/debian-user/2023/06/msg00009.html).

**Reply by tsujan** [reply]
> Do you mean the following scenario by "stacking WMs": Start Menu ⟶ Sub-menus ⟶ possibly further sub-sub-menus) ⟶ A particular Program Link with every link on the list?

I may haven't understood this question. WM has no relation to start menu. *Some* WMs may provide desktop right-click menus, but such menus will be shown only if you stop pcmanfm-qt 's desktop module, in which case you'll lose desktop icons and all other desktop features provided by pcmanfm-qt.

The term "start menu" or "main menu" is usually used for the application menu of LXQt Panel. In LXQt, desktop isn't a place for showing an app menu, as it isn't in KDE or GNOME.

> Does xfwm accompany xfce or just related to xfce? Does KWin accompany KDE or just related to KDE?

xfwm4 and KWin are parts of XFCE and KDE respectively, but they can be used without/outside XFCE or KDE. The same is true for LXQt, e.g., pcmanfm-qt is a part of LXQt but can be used without/outside LXQt. This is called "modularity", which is a good thing in Linux/Unix — as long as GNOME is ignored.

**Reply by tsujan** [reply]
> What is a compositor that you have mentioned in respect of OpenBox?

Most people use picom with Openbox, I think.

WARNING: As said before, it should be disabled/removed before another WM that has a compositor is used.

As for the difference between stacking and tiling WMs see this [Wikipedia article](https://en.wikipedia.org/wiki/Stacking_window_manager).

**Reply by rajibando** [reply]
Thank you, Mr. [tsujan](https://github.com/tsujan), for replying to my queries.
Understood most of your points.
After your post I checked [wikipedia.org › wiki › Stacking_window_manager](https://en.wikipedia.org/wiki/Stacking_window_manager) to know about Stacking WMs. Again the term 'compositing algorithm' came into pic. In Debian synaptic, picom has the following entry:

> picom is a compositor for X11, based on xcompmgr. In addition to shadows,
fading and translucency, picom implements window frame opacity control,
inactive window transparency, and shadows on argb windows.

So this part also doesn't remain alien to me. 

I would request you to address the points on [my other post here](https://github.com/lxqt/lxqt/discussions/2432#discussioncomment-6056143) too.
